### PR TITLE
DPG-1714: Fields stories fixes, core module components integration.

### PR DESCRIPTION
**Pull request by nabeelmd-eGov**

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/package.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     "packages/libraries-core",
     "example",
     "packages/css",
-    "packages/components-core",
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/package.json:10]: revert back this change

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/ButtonSelector.js [hunk 2]
```diff
@@ -20,7 +20,9 @@ const ButtonSelector = (props) => {
       disabled={props.isDisabled}
       style={props.style ? props.style : null}
     >
-      <h2 style={{ ...props?.textStyles, ...{ width: "100%" } }}>{props.label}</h2>
+      <h2 style={{ ...props?.textStyles }} className="digit-button-label">
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/ButtonSelector.js:23]: @nabeelmd-eGov  just see what is difference between button , button selector and submitbar 
you can have a wrapper component but should not use <button> again

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/HeaderBar.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const HeaderBar = (props) => {
+  return (
+    <div className="header-wrap" style={props?.style ? props.style : {}}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/HeaderBar.js:6]: what is App bar and this header bar

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/ButtonSelector.js (hunk 2 above)
Comment (nabeelmd-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/ButtonSelector.js]: @jagankumar-egov replaced with button component

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/HeaderBar.js (hunk 3 above)
Comment (nabeelmd-eGov) [reply frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/HeaderBar.js]: @jagankumar-egov replaced with header component

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/InputCard.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  inputs = [],
+  inputRef,
+  onNext,
+  onSkip,
+  isDisable,
+  onAdd,
+  isMultipleAllow = false,
+  cardStyle = {},
+}) => {
+  const isMobile = window.Digit.Utils.browser.isMobile();
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/InputCard.js:26]: write media query

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/DetailsCard.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            key={itemIndex}
+            style={{ border: selectedItems?.includes(object[keyForSelected]) ? "2px solid #f47738" : "2px solid #fff" }}
+            className="details-container"
+            onClick={() =>handleClickEnabled && handleSelect(object)}
+          >
+            {Object.keys(object).filter(rowEle => !(typeof object[rowEle] == "object" && object[rowEle]?.hidden == true)).map((name, index) => {
+              return <Details label={name} name={object[name]} key={index} onClick={() =>handleClickEnabled && handleDetailCardClick(object)} />;
+            })}
+            {showCitizenInfoLabel ?<CitizenInfoLabel
+              style={{ margin: " 2rem 0px", padding: "10px", backgroundColor: "#FFE2B5", borderRadius: "0.25rem" }}
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/DetailsCard.js:68]: remove inline styles

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/DetailsCard.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          return (
+            <Link
+              key={itemIndex}
+              to={isTwoDynamicPrefix 
+                ?
+                  `${linkPrefix}${typeof serviceRequestIdKey === "function"
+                    ?
+                    serviceRequestIdKey(object)
+                      :
+                    `${getRedirectionLink(object["Application Type"]==="BPA_STAKEHOLDER_REGISTRATION"?"BPAREG":"BPA")}/${object[object["Application Type"]==="BPA_STAKEHOLDER_REGISTRATION"?"applicationNo":"Application Number"]}`}`
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/DetailsCard.js:32]: remove mission specific code

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/DetailsCard.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Link } from "react-router-dom";
+import CitizenInfoLabel from "../atoms/CitizenInfoLabel";
+import  ActionBar from "../atoms/ActionBar";
+import Button from "../atoms/Button";
+export const Details = ({ label, name, onClick}) => {
+  return (
+    <div className="detail" onClick={onClick}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/molecules/DetailsCard.js:9]: class names

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/hoc/Modal.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  const mobileView = Digit.Utils.browser.isMobile() ? true : false;
+  useEffect(() => {
+    document.body.style.overflowY = "hidden";
+    return () => {
+      document.body.style.overflowY = "auto";
+    };
+  }, []);
+  return (
+    <PopUp>
+      <div className="popup-module" style={popupStyles}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/hoc/Modal.js:40]: class names

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/WhatsNewCard.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+const WhatsNewCard = ({header, actions, eventNotificationText, timePastAfterEventCreation, timeApproxiamationInUnits}) => {
+    const { t } = useTranslation();
+
+    const getTransformedLocale = label => {
+        if (typeof label === "number") return label;
+        return label && label.toUpperCase().replace(/[.:-\s\/]/g, "_");
+    };
+
+    return <div className="WhatsNewCard">
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/WhatsNewCard.js:12]: class names

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/PageBasedInput.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import Card from "../atoms/Card";
+import Button from "./Button";
+
+const PageBasedInput = ({ children, texts, onSubmit }) => {
+  return (
+    <div className="PageBasedInputWrapper PageBased">
+      <Card>
+        {children}
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/PageBasedInput.js:9]: classname

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/CitizenHomeCard.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+const CitizenHomeCard = ({ header, links = [], state, Icon, Info, isInfo = false, styles }) => {
+  return (
+    <div className="CitizenHomeCard" style={styles ? styles : {}}>
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/CitizenHomeCard.js:6]: classname

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/CardSubHeader.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+const CardSubHeader = (props) => {
+  const user_type = Digit.SessionStorage.get("user_type") === "employee" ? true : false;
+
+  return (
+    <header
+      className={`${user_type ? "employee-card-sub-header" : "card-sub-header"} ${props?.className ? props?.className : ""}`}
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/CardSubHeader.js:8]: classnames

File: frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/CardSectionHeader.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const CardSectionHeader = (props) => {
+  return (
+    <header id={props.id} className="card-section-header" style={props.style}>
+      {props.children}
```
Comment (jagankumar-egov) [inline frontend/micro-ui/web/micro-ui-internals/packages/components-core/src/atoms/CardSectionHeader.js:6]: make all card header, subheader as a generic , card header component and use different class names which or a variant way.

**Review by jagankumar-egov — APPROVED** [review]

**Review by jagankumar-egov — CHANGES_REQUESTED** [review]
check again @nabeelmd-eGov

**Review by jagankumar-egov — APPROVED** [review]

**Pull request merged**